### PR TITLE
Allow copying custom attributes when replacing an element with the `inline` plugin

### PR DESCRIPTION
**Pull request by bglw**
## Description

Opening a PR to start discussion on this, I've intentionally left the test failing to illustrate the changes.

I have a use case where I need to set attributes on an element that is output using the `inline` plugin. For example, using AlpineJS to show or hide an SVG that I'm inlining:

```html
<img class="image" src="my.svg" x-show="flag" />
```

Currently outputs:
```html
<svg class="image> . . . </svg>
```

Notably missing the `x-show` attribute.

This PR experiments with migrating _all_ attributes by default, and stripping only the problematic ones (e.g. `<svg src="">`. An open question I have is how exhaustive the stripping of these attributes should be. For example, `crossorigin` is an attribute that can be set on a `<link>` tag, and is meaningless on a `<style>` tag. Enumerating all of these attributes is something that could be done, but I don't know what the utility of setting them on the pre-inlined element would be in the first place.

One thought would be adding an option for an allow-list of attributes to pass through, but due to the nature of alpine attributes (or `data-*` tags) this doesn't seem plausible to offload to the user's configuration.

Keen to hear your thoughts on the direction of this, happy to make any changes as needed. I would really love a way to achieve the goal referenced above, but this PR doesn't have to be the implementation that provides that.

## Related Issues
—

### Check List

- [x] Have you read the
      [CODE OF CONDUCT](https://github.com/lumeland/lume/blob/master/CODE_OF_CONDUCT.md)
- [x] Have you read the document
      [CONTRIBUTING](https://github.com/lumeland/lume/blob/master/CONTRIBUTING.md)
  - [x] One pull request per feature. If you want to do more than one thing,
        send multiple pull request.
  - [x] Write tests.
  - [x] Run deno `fmt` to fix the code format before commit.
  - [x] Document any change in the `CHANGELOG.md`.

### CODE REVIEW

**Comment by oscarotero** [comment]
I don't think passing all attributes is a good idea. The option for an allow-list of attributes sounds good. It could accept strings or regular expressions for advanced use cases like `data-*` and `x-*`. For example:

```ts
site.use(inline({
  copyAttributes: ["title", /^data-/, /^x-/]
}));
```

This only affects to when new elements are created (`<link>` to `<style>` and `` to `<svg>`), so not sure if a list for each case is better, because some attributes make sense only in one of these cases.

```ts
site.use(inline({
  styleAttributes: [],
  svgAttributes: []
}));
```

What do you think?

**Comment by bglw** [comment]
Fab, regular expressions gets my vote. 

I don’t think separate lists are needed - it’s getting pretty niche if you’re writing data-* attributes on your <link> tags that you don’t want. If they didn’t make sense there, I can’t see why you would write them in the first place. 

I’ll amend this PR and implement the copyAttributes regex.

**Comment by bglw** [comment]
@oscarotero Ready for review 👍

File: plugins/inline.ts [hunk 1]
```diff
@@ -200,14 +206,8 @@ export default function (userOptions?: Partial<Options>) {
...
-            // Width and height attributes
-            if (element.getAttribute("width")) {
-              svg.setAttribute("width", element.getAttribute("width")!);
-            }
-
-            if (element.getAttribute("height")) {
-              svg.setAttribute("height", element.getAttribute("height")!);
-            }
+            migrateAttributes(element, svg, ["id", "class", "width", "height"]);
+            svg.removeAttribute("src");
```
Comment (oscarotero) [inline plugins/inline.ts:210]: Is this really necessary? SVG code shouldn't have a `src` attribute.

**Comment by oscarotero** [comment]
@bglw Everything looks great! I just left a comment.

File: plugins/inline.ts (hunk 1 above)
Comment (bglw) [reply plugins/inline.ts:210]: Oops no, good spot. Just a remnant from the previous iteration. I can delete that tomorrow (or happy for you to do so)

Comment (oscarotero) [reply plugins/inline.ts]: Okay, understood. I just fixed it. Thanks!

**Pull request merged**